mantatail.py

```diff
@@ -202,14 +202,11 @@ class UserConnection:
         A Tuple containing (None, None) indicates a QUIT command and closes the connection to the client.
     """
 
-    # self.nick is defined in recv_loop()
-    # It is not set in __init__ to keep mypy happy.
-    nick: str  # Nick is shown in user lists etc, user_name is not
-
     def __init__(self, state: ServerState, host: str, socket: socket.socket):
         self.state = state
         self.socket = socket
         self.host = host
+        self.nick: str = "*"
```

Comment:
Mypy can now correctly guess that `nick` is a string. There's no need to tell it to mypy manually.

```suggestion
        self.nick = "*"
```